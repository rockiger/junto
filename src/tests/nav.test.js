import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router-dom';

import Nav from '../components/nav';

describe('Nav', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <MemoryRouter>
                <Nav />
            </MemoryRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <MemoryRouter>
                <Nav />
            </MemoryRouter>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});