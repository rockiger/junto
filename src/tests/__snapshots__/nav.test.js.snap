// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav has a valid snapshot 1`] = `
<nav
  className="Nav"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <span
      className="Nav-logo"
    >
      <img
        alt="Logo"
        src="logo.svg"
      />
      <span
        className="Nav-logo-name"
      >
        Awiki
      </span>
    </span>
  </a>
  <div
    className="Nav-search"
  />
  <div
    className="Nav-user"
  />
  <style>
    
            :global(body) {
                margin: 0;
                font-family: 'Open Sans', -apple-system, BlinkMacSystemFont,
                    Avenir Next, Avenir, Helvetica, sans-serif;
            }
            :global(main) {
                margin-top: 64px;
            }
            .Nav {
                position: fixed;
                top: 0;
                display: flex;
                padding: 8px 16px;
                font: 13px/27px Roboto, RobotoDraft, Arial, sans-serif;
                font-family: 'Google Sans', Roboto, Arial, sans-serif;
                flex-grow: 1;
                flex-shrink: 0;
                flex-basis: auto;
                border-bottom: 1px solid #dadce0;
                width: 100%;
                box-sizing: border-box;
                background-color: white;
                z-index: 1000;
            }
            .Nav-logo {
                height: 48px;
                min-width: 238px;
                flex-grow: 1;
                flex-shrink: 0;
                flex-basis: auto;
                display: flex;
                align-items: center;
            }
            .Nav-logo-name {
                display: inline-block;
                font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI',
                    Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans',
                    'Helvetica Neue', sans-serif;
                font-size: 22px;
                vertical-align: middle;
                color: #5f6368;
                line-height: 24px;
                padding-left: 4px;
            }
            .Nav-search {
                display: flex;
                height: 48px;
                flex-grow: 1;
                flex-shrink: 1;
                flex-basis: 100%;
            }
            .Nav-search-form {
                background: #f1f3f4;
                border: 1px solid transparent;
                -webkit-border-radius: 8px;
                border-radius: 8px;
                display: flex;
                flex: 1 1 auto;
                margin-right: auto;
                max-width: 720px;
                padding-left: 10px;
                padding-right: 30px;
                position: relative;
                -webkit-transition: background 100ms ease-in,width 100ms ease-out;
                transition: background 100ms ease-in,width 100ms ease-out;
            }
            .Nav-search-form input {
                background-color: #f1f3f4;
                border: none;
                font-size: 1rem;
                margin-left: 46px;
            }
            .Nav-user {
                height: 48px;
                flex-grow: 0;
                flex-shrink: 0;
                flex-basis: auto;
                display: flex;
                justify-content: center;
                flex-direction: column;
            }
            .Nav ul {
                display: flex;
                justify-content: space-between;
            }
            .Nav &gt; ul {
                padding: 4px 16px;
            }
            .Nav li {
                display: flex;
                padding: 6px 8px;
            }
            .Nav a {
                color: #067df7;
                text-decoration: none;
                font-size: 13px;
            }
        
  </style>
</nav>
`;
